Updating broken/outdated links in the ProgrammerManual

jmriesen opened the pull request:
Fixes llvm/llvm-project#117897

jmriesen committed:
Updating broken/outdated links in the ProgrammerManual

fmayer commented on llvm/docs/ProgrammersManual.rst:
```diff
@@ -3358,14 +3358,14 @@ the ``PassManager.h`` system, and there is a more detailed introduction to it
 #. `Sean Parent's Papers and Presentations
-   <http://github.com/sean-parent/sean-parent.github.com/wiki/Papers-and-Presentations>`_
+   <https://sean-parent.stlab.cc/papers-and-presentations>`_
    - A GitHub project full of links to slides, video, and sometimes code.
```
it is no longer a github project

jmriesen committed:
Updating link description

Updating Sean Parent's Papers and Presentations link description since
it no longer points to a GitHub project.

fmayer reviewed (approved):
thanks
